x-pack/plugins/security_solution/public/common/components/exceptions/viewer/index.tsx

```diff
@@ -4,96 +4,411 @@
+  overflow: hidden;
 `;
 
-interface ExceptionItemProps {
-  exceptionItem: ExceptionListItemSchema;
+const initialState: State = {
+  filterOptions: { filter: '', showEndpointList: false, showDetectionsList: false, tags: [] },
+  pagination: {
+    pageIndex: 0,
+    pageSize: 20,
+    totalItemCount: 0,
+    pageSizeOptions: [5, 10, 20, 50, 100, 200, 300],
+  },
+  endpointList: null,
+  detectionsList: null,
+  allExceptions: [],
+  exceptions: [],
+  exceptionToEdit: null,
+  loadingItemIds: [],
+  isModalOpen: false,
+};
+
+enum ModalAction {
+  CREATE = 'CREATE',
+  EDIT = 'EDIT',
+}
+
+interface ExceptionsViewerProps {
+  ruleId: string;
+  exceptionListsMeta: ExceptionIdentifiers[];
+  availableListTypes: ExceptionListType[];
   commentsAccordionId: string;
-  handleDelete: ({ id }: { id: string }) => void;
-  handleEdit: (item: ExceptionListItemSchema) => void;
+  onAssociateList?: (listId: string) => void;
 }
 
-const ExceptionItemComponent = ({
-  exceptionItem,
+const ExceptionsViewerComponent = ({
+  ruleId,
+  exceptionListsMeta,
+  availableListTypes,
+  onAssociateList,
   commentsAccordionId,
-  handleDelete,
-  handleEdit,
-}: ExceptionItemProps): JSX.Element => {
-  const [entryItems, setEntryItems] = useState<FormattedEntry[]>([]);
-  const [showComments, setShowComments] = useState(false);
+}: ExceptionsViewerProps): JSX.Element => {
+  const { services } = useKibana();
+  const [, dispatchToaster] = useStateToaster();
+  const [initLoading, setInitLoading] = useState(true);
+  const onDispatchToaster = useCallback(
+    ({ title, color, iconType }) => (): void => {
+      dispatchToaster({
+        type: 'addToaster',
+        toast: {
+          id: uuid.v4(),
+          title,
+          color,
+          iconType,
+        },
+      });
+    },
+    [dispatchToaster]
+  );
+  const { deleteExceptionItem } = useApi(services.http);
+  const [
+    {
+      endpointList,
+      detectionsList,
+      exceptions,
+      filterOptions,
+      pagination,
+      loadingItemIds,
+      isModalOpen,
+    },
+    dispatch,
+  ] = useReducer(allExceptionItemsReducer(), initialState);
 
-  useEffect((): void => {
-    const formattedEntries = getFormattedEntries(exceptionItem.entries);
-    setEntryItems(formattedEntries);
-  }, [exceptionItem.entries]);
+  const setExceptions = useCallback(
+    ({
+      lists: newLists,
+      exceptions: newExceptions,
+      pagination: newPagination,
+    }: SetExceptionsProps) => {
+      dispatch({
+        type: 'setExceptions',
+        lists: newLists,
+        exceptions: newExceptions,
+        pagination: newPagination,
+      });
+    },
+    [dispatch]
+  );
+  const [loadingList, , , , fetchList] = useExceptionList({
+    http: services.http,
+    lists: exceptionListsMeta,
+    filterOptions,
+    pagination: {
+      page: pagination.pageIndex + 1,
+      perPage: pagination.pageSize,
+      total: pagination.totalItemCount,
+    },
+    dispatchListsInReducer: setExceptions,
```

Comment:
NOTE: I was unable to figure out how to get rid of this typing error. It's due to the use of the differing exceptions structure. Tried type casting as a temporary workaround and that didn't seem to get it to go away.

Reply:
It's super icky, but for now this worked `exceptions: (newExceptions as unknown) as ExceptionListItemSchema[]`. Have a to do to clean up once api updates.